### PR TITLE
Allow module to return success when dealing with 204 response from hi…

### PR DESCRIPTION
**Pull request by guillermo-menjivar**
…pchat v2

### What does this PR do?
PR allows hipchat module to support the 204 response from hipchat v2 interface when submitting a message.

### Previous Behavior
When submitting a message to hipchat v2 it will return a failed state even when the message was successfully submitted and published by hipchat.

### New Behavior
When submitting a message to hipchat v2 it returns a success.

### Tests written?

Yes

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/hipchat.py [hunk 1]
```diff
@@ -176,8 +176,9 @@ def _query(function,
     if result.get('status', None) == salt.ext.six.moves.http_client.OK:
         response = hipchat_functions.get(api_version).get(function).get('response')
         return result.get('dict', {}).get(response, None)
-    elif result.get('status', None) == salt.ext.six.moves.http_client.NO_CONTENT:
-        return False
+    elif result.get('status', None) == salt.ext.six.moves.http_client.NO_CONTENT and \
+        api_version == 'v2':
```
Comment (rallytime) [inline salt/modules/hipchat.py]: How will this change affect those running on version 1?

**Review by guillermo-menjivar — COMMENTED** [review]
This should not affect v1. This is an exclusive conditional for v2. We currently return False if the response is not 200. So if v1 gets a non 200 it will receive the False returned by the else branch.

**Review by rallytime — APPROVED** [review]
Thanks for the explanation! 👍

**Comment by rallytime** [comment]
re-run py2

**Pull request merged**